KEYCLOAK-17920 Allow skipping of resolve attribute from the groups of the user in UserAttributeStatementMapper and UserAttributeMapper

bohmber opened the pull request:
Unexpected call to user.groups if 'Aggregate attribute values' is false and the user attribute is empty in KeycloakModelUtils.resolveAttribute(UserModel user, String name, boolean aggregateAttrs) used by UserAttributeStatementMapper and UserAttributeMapper

This PR is for https://issues.redhat.com/browse/KEYCLOAK-17920

mposolda commented:
@bohmber Thanks for the PR. I see there are some test failures and unfortunately looks like regression of your changes.

I understand that the tooltip for "Aggregate attribute values" does not exactly respect the behaviour, but I think we probably can't simply change the behaviour and remove fallback to looking attribute values to groups in case that attribute is unavailable on user...

Could you please start a discussion on keycloak-dev mailing list to start discussion with wider team about best approach here?

bohmber committed:
KEYCLOAK-17920
Allow skipping of resolve attribute from the groups of the user in UserAttributeStatementMapper and UserAttributeMapper

bohmber commented:
Based on the discussion change the pr to support skipping the resolve attributes in groups

bohmber committed:
KEYCLOAK-17920
Allow skipping of resolve attribute from the groups of the user in UserAttributeStatementMapper and UserAttributeMapper
